.github/workflows/main-checks.yml

```diff
@@ -164,3 +164,72 @@ jobs:
         env:
           RUSTFLAGS: --cfg nightly_yew --cfg yew_lints
         run: cargo test -p yew-macro test_html_lints
+
+  unit_tests_wasi:
+    name: Unit Tests (WASI) on ${{ matrix.toolchain }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - 1.76.0
+          - stable
+          - nightly
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          target: wasm32-wasi
+
+      - name: Install wasmtime
+        run: |
+          wget https://github.com/bytecodealliance/wasmtime/releases/download/v15.0.1/wasmtime-v15.0.1-x86_64-linux.tar.xz
+          tar xf wasmtime-v15.0.1-x86_64-linux.tar.xz
+          mv wasmtime-v15.0.1-x86_64-linux/wasmtime ~/wasmtime
+          rm -rf wasmtime-v15.0.1-x86_64-linux.tar.xz wasmtime-v15.0.1-x86_64-linux
+          chmod +x ~/wasmtime
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Run WASI tests for yew
+        run: |
+          RUST_LOG=info
+          cargo test --features ssr,hydration --target wasm32-wasi -p yew --no-run 2>&1 |\
+            grep -o "target/wasm32-wasi/debug/deps/[^ ]*\.wasm" |\
+            xargs -I {} ~/wasmtime -W unknown-imports-trap=y {}
```

Comment:
> You can use a test runner config in `.cargo/config.toml` here
> 
> ```
> [target.'cfg(all(target_arch = "wasm32", target_os = "wasi"))']
> runner = 'wasmtime -W unknown-imports-trap=y'
> ```
> 
> like we do for the `wasm-bindgen` harness, I believe. This should avoid having to get, parse and run the test cases manually.
> 
> Same line, can you explain a bit what the goal of `unknown-imports-trap=y` is? I'm not super aware of the defaults in wasmtime.

Regarding the extra parameter, this is because the newer version of `wasmtime` adds a security check that requires all imported functions to exist by default.`wasm-bindgen` may generate some global utility functions that do not belong to WASI (such as asking the browser for basic information about the heap, swapping strings, etc.)